.travis.yml

```diff
@@ -59,7 +59,7 @@ before_install:
 
 stages:
   - name: build
-    if: branch = master OR branch =~ ^(.*-test-deploy)$
+    if: branch = master OR branch = test-deploy
```

Comment:
I am asking for these changes because I want consistency in our projects and if you merge these changes to `.travis.yml` I will need to go in and undo them, so please don't do that. If my other comments make no sense to you and you don't understand the issue we can discuss it today.
Thanks!

Reply:
Nathan,

I am lack of knowledge on understanding the details of the workflow that may affect you.  Here is my understanding. 
 This merge will make the potter projects fail due to the change in .yaml in the test_deploy. 
 Here is the correct procedure based on your description.  Please correct me if  I am wrong.  I also have questions, please answer.  
1) I will create my own branch "foo". This should be based on the master's yaml , am I right here? 
2) I want to test. so I create a child branch of foo: foo-test-deploy, I should change the yaml file of foo-test-deploy based on the branch test-deploy. Am I right here?
3) Then I push this foo-test-deploy to github as a branch, as you said, this will trigger Travis to test.
4) If everything works, I delete foo-test-deploy, issue a PR for branch "foo"
5) Ask you to review again.
6) If everything works, merge "foo" to "master".

Reply:
> 1. I will create my own branch "foo". This should be based on the master's yaml , am I right here?

In this example the name `foo` is just a stand-in for some name that actually relates to the work.

> 2. I want to test. so I create a child branch of foo: foo-test-deploy, I should change the yaml file of foo-test-deploy based on the branch test-deploy but change if: branch = master OR branch = test-deploy. to if: branch = master OR branch = foo-test-deploy, Am I right here?

That is not at all what I meant. The point is you did/do not need to change the `.travis.yml` to get what you want. It's already correct. _The master branch version is correct. Please do not change it._

Here's how it works, and why it did not work for your `test-deploy` branch:

- On line 62 of the `.travis.yml` file:
```
if: branch = master OR branch = test-deploy
```
The second clause utilizes regex expression (indicated by th `=~`):  `branch =~ ^(.*-test-deploy)$` which matches ANY branch whose name ends in `-test-deploy` Your (this) branch name `test-deploy` did not trigger a build/deploy because it does not match the expression (note the leading `-`). If you make a branch `foo-test-deploy` it will work as intended. (Instead of `foo` a more descriptive name might be `netcdf-api-mod` and `netcdf-api-mod-test-deploy`, but I digress...)

> 3. Then I push this foo-test-deploy to github as a branch, as you said, this will trigger Travis to test.
> 4. If everything works, I delete foo-test-deploy, issue a PR for branch "foo"

Well, you can issue the PR anytime. I would suggest that, instead of closing a PR so you do more work on the code, you could just mark it as Draft. That will allow you to continue to work, and the CI/CD machine will continue to test it when youngish. But, the notifications will be silenced so reviewers are not informed about every little thing.
 
> 5. Ask you to review again.

👍 

> 6. If everything works, merge "foo" to "master".

👍 


Does that make sense to you?

N

Reply:
Got it. Thanks much.

Reply:
I decide to close this PR to follow the procedure from scratch.